sorc/global_cycle.fd/land_increments.f90

```diff
@@ -361,7 +381,163 @@ subroutine add_increment_soil(rla,rlo,stc_state,smc_state,slc_state,stc_updated,
 
     deallocate(id1, id2, jdc, s2c)
 
-end subroutine add_increment_soil
+end subroutine add_gsi_increment_soil
+
+ !> Read in fv3-jedi file with soil state increments (on the cubed-sphere 
+ !! grid),and to the soil states. Adapted from add_gsi_increment_soil.
+ !!
+ !! @param[in] SLCINC Liquid soil moisture increments on the cubed-sphere tile
+ !! @param[in] STCINC Soil temperature increments on the cubed-sphere tile
+ !! @param[inout] STC_STATE Soil temperature state vector
+ !! @param[inout] SMC_STATE Soil moisture (liquid plus solid) state vector
+ !! @param[inout] SLC_STATE Liquid soil moisture state vector
+ !! @param[out] stc_updated Integer to record whether STC in each grid cell was updated 
+ !! @param[out] slc_updated Integer to record whether SMC in each grid cell was updated 
+ !! @param[in] SOILSNOW_TILE Land mask for increments on the cubed-sphere tile
+ !! @param[in] SOILSNOW_FG_TILE First guess land mask for increments on the
+ !!             cubed-sphere tile
+ !! @param[in] LENSFC Number of land points on a tile
+ !! @param[in] LSOIL Number of soil layers
+ !! @param[in] lsm Integer flag indicating which land model is used 
+ !! @param[in] MYRANK MPI rank number
+ !!
+ !! @author Yuan Xue. 11/2023
+
+subroutine add_jedi_increment_soil(stcinc,slcinc,stc_state,smc_state,slc_state,stc_updated,&
```

Comment:
is the logic for adding the increments exactly the same as for the GSI option?

Reply:
Very similar, but have small discrepancies. Will discuss via a follow-up meeting.